File: Python3/프로그래머스/2/64065. 튜플/튜플.py
import re

def solution(s):
    arr = []
    answer = []
    
    s = s[2 : -2].split("},{")
    for str in s:
        str = list(map(int, str.split(',')))
        arr.append(str)
    arr.sort(key = lambda obj: len(obj))
    
    for tup in arr:
        for val in tup:
            if val not in answer:
                answer.append(val)
                break
                
    return answer

File: JavaScript/프로그래머스/2/64065. 튜플/튜플.js
function solution(s) {
    const allVal = [...s.match(/(\d+)/g)];
    const map = new Map();
    
    for (let i of allVal) {
        map.set(i, (map.get(i) || 0) + 1);
    };
    
    const sorted = [...map.entries()].sort((a, b) => b[1] - a[1]);    
    var answer = sorted.map(([key, value]) => Number(key));
    return answer;
}